SPIE2016-9913-16/9913-16.tex

```diff
@@ -0,0 +1,303 @@
+LSST has its own spherical geometry package in C++, \texttt{lsst::sphgeom}, and we should compare implementations with the STScI spherical geometry library (mainly written in Python) that is being proposed for integration into Astropy.
+There is currently a focus within Astropy on handling region files from the DS9 application\cite{2005ASPC..347..110J} but there have also been discussions concerning adopting the IVOA STC standard\cite{2007ivoa.spec.1030R} as a region serialization format.
+STC-S is already being used to define sky regions in the VO community and is also understood by the GAIA visualization tool\cite{2009ASPC..411..575D} and Starlink software\cite{2010ASPC..434..213B,2014ASPC..485..391C}.
+If Astropy and LSST cannot share sky to pixel region handling code directly there is a possibility that STC could be adopted to allow interoperability.
+
+\begin{table}[t]
+\centering
+\caption{Four distinct types of region. This paper discusses all except discrete spherical.}
+\label{tab:regions}
+\vspace{10pt}
+\begin{tabular}{ccc}
+  Discrete Spherical & $\Longleftrightarrow$ & Discrete Cartesian \\
+  $\Updownarrow$ & &  $\Updownarrow$ \\
+  Geometric Spherical & $\Longleftrightarrow$ & Geometric Cartesian \\
+\end{tabular}
+\end{table}
+
+Astropy has no plans for a discretized efficient representation of a pixel region and this is something that LSST can contribute to the Astropy region handling package.
+In LSST a \emph{footprint} is a compact serialized representation of a mask, listing the pixel indices that contribute to the shape in rasterized order, using a run-length encoding algorithm to conserve space.
+A \emph{heavy footprint} is a \emph{footprint} that includes the pixel values themselves.
+These are the items that can be passed to measurement algorithms.
+\emph{Footprints} are used extensively in the LSST code base and would be a useful addition to Astropy.
+
+There are no plans currently in either project to handle sky pixelization representations of regions, such as multi-order coverage maps (MOC)\cite{2012ASPC..461..347F,2014ivoa.spec.0602F} or other approaches\cite{2015A&A...580A.132R}.
+
+\subsection{LSST Measurement Algorithms}
+
+There are a number of measurement algorithms in LSST that could be made available as Astropy affiliate packages\cite{bosch_2016_48435}.
+The initial idea is to release the LSST equivalents of a subset of the Astromatic suite\cite{2012ASSP...29...71B}.
+This would have the advantage of providing a simple way to test the process for migrating native LSST measurement algorithm code to the Astropy ecosystem in a form that is easily validatable by the community.
+Once complete it should then be possible to consider migrating the more advanced LSST algorithms.
+Making them available in this way should aid adoption by the wider community and provide early feedback as to the quality of the algorithms and detect any bugs or anomalous behavior well before commissioning of LSST begins.
```

Comment:
I don't like calling these just "measurement algorithms" (as in my comment on the description of `meas`, I think there's quite a bit more that we should explicitly mention).  And I think we should explain that this is both the highest-value contribution LSST could make to the Astropy ecosystem and the most difficult one to accomplish, since it essentially depends on all the lower-level code being interoperable first.

Mind if I just take a stab at rewriting this paragraph?